plugin/fzf.vim

```diff
@@ -26,10 +26,62 @@ if exists('g:loaded_fzf')
 endif
 let g:loaded_fzf = 1
 
+function! fzf#shellescape(path)
+  if s:is_win
+    let shellslash = &shellslash
+    try
+      set shellslash
+      return '"'.shellescape(a:path).'"'
+    finally
+      let &shellslash = shellslash
+    endtry
+  endif
+  return shellescape(a:path)
+endfunction
+
+function! fzf#getcwd()
+  if s:is_win
+    let shellslash = &shellslash
+    try
+      set noshellslash
+      return getcwd()
+    finally
+      let &shellslash = shellslash
+    endtry
+  endif
+  return getcwd()
+endfunction
+
+function! fzf#fnamemodify(fname, mods)
+  if s:is_win
+    let shellslash = &shellslash
+    try
+      set noshellslash
+      return fnamemodify(a:fname, a:mods)
+    finally
+      let &shellslash = shellslash
+    endtry
+  endif
+  return fnamemodify(a:fname, a:mods)
+endfunction
+
+function! fzf#expand(fmt)
+  if s:is_win
+    let shellslash = &shellslash
+    try
+      set noshellslash
+      return expand(a:fmt)
+    finally
+      let &shellslash = shellslash
+    endtry
+  endif
+  return expand(a:fmt)
+endfunction
+
```

Comment:
I used `fzf#` prefix because I thought that [fzf.vim] can use it to handle `shellslash`. However, these new functions are  not suppose to change and are integral to [fzf]'s Vim/Neovim plugin. Outside of [fzf] and [fzf.vim], users shouldn't rely on these new functions. Do you prefer to minimize Vimscript dependencies between [fzf] and [fzf.vim]?

[fzf]: https://github.com/junegunn/fzf
[fzf.vim]: https://github.com/junegunn/fzf.vim